lnwallet/test/test_interface.go

```diff
@@ -3467,3 +3530,339 @@ func runTests(t *testing.T, walletDriver *lnwallet.WalletDriver,
+// enrichment.
+type psbtChecker struct {
+	*btcwallet.BtcWallet
+
+	t *testing.T
+
+	checker *input.DefaultSignDescriptorChecker
+
+	chanMtx sync.Mutex
+
+	channelCheckers []func(*wire.TxOut, map[string][]byte) error
+
+	seenTxs map[chainhash.Hash]struct{}
+
+	addedChannel bool
+}
+
+func newPsbtChecker(t *testing.T, wallet *btcwallet.BtcWallet) *psbtChecker {
+	c := &psbtChecker{
+		wallet,
+		t,
+		&input.DefaultSignDescriptorChecker{},
+		sync.Mutex{},
+		[]func(*wire.TxOut, map[string][]byte) error{},
+		make(map[chainhash.Hash]struct{}),
+		false,
+	}
+
+	c.checker.OutChecker = c.outChecker
+
+	return c
+}
+
+func (c *psbtChecker) outChecker(out *wire.TxOut,
+	unkMap map[string][]byte) error {
+
+	// Check for funding output.
+	var (
+		remoteKey, localKey *btcec.PublicKey
+		err                 error
+	)
+
+	remoteKeyBytes, hasRemote := unkMap[string(
+		input.PsbtKeyTypeOutputRemoteMultiSigKey,
+	)]
+	if hasRemote {
+		remoteKey, err = btcec.ParsePubKey(remoteKeyBytes)
+		if err != nil {
+			return err
+		}
+	}
+
+	localDescBytes, hasLocalDesc := unkMap[string(
+		input.PsbtKeyTypeOutputLocalMultiSigKey,
+	)]
+	if hasLocalDesc {
+		intWallet := c.InternalWallet()
+
+		fingerprint, coin, localDesc, err :=
+			input.KeyDescriptorFromUnknownValue(localDescBytes)
+		if err != nil {
+			return err
+		}
+
+		scope, account, err := intWallet.LookupAccount("default")
+		if err != nil {
+			return err
+		}
+
+		props, err := intWallet.AccountProperties(scope, account)
+		if err != nil {
+			return err
+		}
+
+		if props.MasterKeyFingerprint != fingerprint {
+			return fmt.Errorf("fingerprint doesn't match: "+
+				"expected %d, got %d",
+				props.MasterKeyFingerprint, fingerprint)
+		}
+
+		if coin != netParams.HDCoinType {
+			return fmt.Errorf("coin type doesn't match: "+
+				"expected %d, got %d", netParams.HDCoinType,
+				coin)
+		}
+
+		keyRing := keychain.NewBtcWalletKeyRing(intWallet, coin)
+		derivedDesc, err := keyRing.DeriveKey(localDesc.KeyLocator)
+		if err != nil {
+			return err
+		}
+
+		if localDesc.PubKey != nil &&
+			!localDesc.PubKey.IsEqual(derivedDesc.PubKey) {
+
+			return fmt.Errorf("local descriptor pubkey mismatch: "+
+				"passed %x, derived %x",
+				localDesc.PubKey.SerializeCompressed(),
+				derivedDesc.PubKey.SerializeCompressed())
+		}
+
+		localKey = derivedDesc.PubKey
+	}
+
+	if hasLocalDesc && hasRemote {
+		witness, err := input.GenMultiSigScript(
+			localKey.SerializeCompressed(),
+			remoteKey.SerializeCompressed(),
+		)
+		if err != nil {
+			return err
+		}
+
+		pkScript, err := input.WitnessScriptHash(witness)
+		if err != nil {
+			return err
+		}
+
+		if bytes.Equal(pkScript, out.PkScript) {
+			c.addChannel(unkMap)
+
+			return nil
+		}
+	}
+
+	// Check for change output. A validating remote signer should look
+	// ahead based on known wallet state, but we cheat here and look it
+	// up in our known addresses. We only do this if the output has no
+	// unknowns.
```

Comment:
Is this OK, or should we add another unknown with a descriptor here? We'd need to plumb that through as well, through the `ChangeAddr` function when we initialize the wallet assembler.